### PR TITLE
Rename quark.os.FS.fileContents to readFileAsString

### PR DESCRIPTION
**Pull request by ark3**
Please comment on the API defined in os.q.

https://github.com/datawire/quark/blob/os-api/quarkc/lib/os.q

### CODE REVIEW

**Comment by rhs** [comment]
I think we should adopt the getFoo/setFoo convention for all getters and setters. This would imply userHomeDir should grow a get in front of it.

Also, I think our names should consistently map to the corresponding environment variables, i.e. the environment variable is HOME, so it should be getHome, not getUserHomeDir.

**Comment by ark3** [comment]
What's currently quark.os.FS.userHomeDir() does not use the process environment to get the user's home directory. Rather, it uses a target-language-specific mechanism.

**Comment by rhs** [comment]
The FS and Env class names are kind of cryptic. I'd like to build out these APIs by assuming that what we are providing now are merely convenience functions for a more complete API that will be filled out later. For example I'd suggest maybe FS.readBlah -> File.readBlah where File could later grow more functionality.

For Env I'd suggest rename to Environment. This gives the option to have a full Environment instance that might eventually implement Map<String,String>.

**Comment by plombardi89** [comment]
Do we ever intend to have a File object itself? E.g. a java.io.File which is a reference to an actual File on the filesystem? I'd reserve that name for that if we are planning on it.

I'd recommend quark.io for the package name. Things like platform independent Stream implements, File implements etc should end up in this package.

**Comment by plombardi89** [comment]
For what it's worth Java since NIO was introduced has a class called `Files` which has useful static IO methods on it, for example, reading as lines: 

https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html

**Comment by rhs** [comment]
Oops, one more comment. Should getHome() actually just be implemented the same way getUser() is, i.e. just an environment lookup?

**Comment by plombardi89** [comment]
Does readFileAsString need to be charset aware?

**Comment by ark3** [comment]
@rhs The target languages provide a platform-independent way to get the home directory for compatibility with Windows, I'd guess.
@plombardi89 I don't understand your question.

**Comment by plombardi89** [comment]
@rhs ... not sure that's a good idea. Read the bottom section of this: https://docs.oracle.com/javase/tutorial/essential/environment/env.html

"
There are many subtle differences between the way environment variables are implemented on different systems. For example, Windows ignores case in environment variable names, while UNIX does not. The way environment variables are used also varies. For example, Windows provides the user name in an environment variable called USERNAME, while UNIX implementations might provide the user name in USER, LOGNAME, or both.

To maximize portability, never refer to an environment variable when the same value is available in a system property. For example, if the operating system provides a user name, it will always be available in the system property user.name.
"

**Comment by ark3** [comment]
Of course, we don't really have Windows compatibility anyhow. The tests in os_test.q will not succeed on Windows, for example.

**Comment by plombardi89** [comment]
@ark3 Maybe not. But we shouldn't pretend like Windows developers building stuff do not exist either. They're very very very common and we will need to support them eventually. Point is, we shouldn't box ourselves in under the pretense that we don't support Windows now.

File: quarkc/lib/os.q [hunk 1]
```diff
@@ -20,24 +20,20 @@ namespace quark {
...
-            @doc("expect to come into play do, in fact, come into play.")
-            static FileContents fileContents(String path) {
-                FileContents result = new FileContents();
-                _get_file_contents(path, result);
-                return result;
-            }
+        @doc("Read the entire contents of a file into a String. Assume UTF-8 encoding.")
+        @doc("Returns a FileContents Future instance.")
+        @doc("if result.getError() is null then result.value has the file contents.")
+        @doc("NOTE WELL: all the typical memory constraints you might")
```
Comment (rhs) [inline quarkc/lib/os.q]: Can we kill the NOTE WELL? It doesn't actually provide any information.

**Comment by rhs** [comment]
I'm +1 modulo the "NOTE WELL" comment.

**Comment by kflynn** [comment]
Would y'all have been happier with 

```
NOTE WELL: readFileAsString will utterly fail if the file is too large to fit entirely in memory
```

or some such? Seems worthwhile to remind people of that.

**Pull request merged**